docs/features/hooks.md

```diff
@@ -1,9 +1,41 @@
-# Django Hooks
-
 ???+ tip "Looking for more hooks?"
 
     Check out the [IDOM Core docs](https://idom-docs.herokuapp.com/docs/reference/hooks-api.html?highlight=hooks) on hooks!
 
+## Use Sync to Async
+
+This is the suggested method of performing ORM queries when using Django IDOM. 
+
+Fundamentally, this hook is an ORM-safe version of [use_effect](https://idom-docs.herokuapp.com/docs/reference/hooks-api.html#use-effect).
+
+```python title="components.py"
+from example_project.my_app.models import Category
+from channels.db import database_sync_to_async
+from idom import component, html
+from django_idom import hooks
+
+@component
+def simple_list():
+    categories, set_categories = hooks.use_state(None)
+
+    @hooks.use_sync_to_async
+    def get_categories():
+        if categories:
+            return
+        set_categories(Category.objects.all())
+
+    if not categories:
+        return html.h2("Loading...")
```

Comment:
Part of the reason for the `data, loading, error` pattern is that the `if not categories` check does not indicate whether the query is loading - it could be that there are no categories to show, that loading categories failed, or that categories are indeed loading. You also need a convenient way to re-execute queries that are related. For example, triggering an `add_categories` database operation should cause the `get_categories` effect to re-run. Whatever solution we provide should make it easier to distinguish between the possible states a query or mutation is in, as well as re-trigger dependent queries.